packages/entity-server/src/entity/Entity.ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable max-classes-per-file */
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { EntityModel as BaseEntityModel } from '@tupaia/database';
+
+export interface EntityFields {
+  readonly code: string;
+  readonly country_code: string;
+}
+
+export interface EntityModel extends BaseEntityModel, EntityFields {
+  isProject: () => boolean;
```

Comment:
Unfortunately typescript can't infer the `isProject()`, etc. functions on the `EntityModel` class... I think because they're actually attached to the `EntityType` class the database model manually exposes them... oh well 🤷

Reply:
Hmmm...I don't think the model _does_ have `isProject()` etc. functions! 

Basically the model is more of a factory, and produces instances of `EntityType` - _those_ are the bit with the instance specific functions.

Do you want to talk about this on a quick call?